girder/web_client/src/views/widgets/ItemListWidget.js

```diff
@@ -88,13 +145,37 @@ var ItemListWidget = View.extend({
             viewLinks: this._viewLinks,
             showSizes: this._showSizes,
             highlightItem: this._highlightItem,
-            selectedItemId: (this._selectedItem || {}).id
+            selectedItemId: (this._selectedItem || {}).id,
+            paginated: this._paginated
+
         }));
 
-        // If we set a selected item in the beginning we will center the selection while loading
         return this;
     },
 
+    /**
+     * @returns {number} the number of pages in the itemList for use in a paginated view
+     */
+    getNumPages() {
+        return this._totalPages || 1;
+    },
+    /**
+     * @returns {number} the current page for paginated lists, defaults to 1 if none is provided
+     */
+    getCurrentPage() {
+        return this.currentPage || 1;
+    },
+    /**
+     * Externally facing function to allow hierarchyWidget and others to set the current page if the item is paginated
+     * @param {Number} page 1 index integer specifying the page to fetch
+     */
+    setPage(page) {
+        if (this._paginated && this.collection && this.collection.fetchPage) {
+            this.currentPage = page;
+            this.$el.parents('.g-hierarchy-widget-container').scrollTop(0);
```

Comment:
Currently, when you change page, this scrolls to the top of the existing page and the fetches the new page.  I think it would be a better experience if it fetched the new page and only scrolled to the top when that data had been fetched.